### ISSUE TITLE
vue-router document page search error

### ISSUE THREAD

**Issue by falstack** [vuejs/vue-router#1738]
### Version
2.7.0

### Reproduction link
[https://router.vuejs.org/en/](https://router.vuejs.org/en/)

### Steps to reproduce
open this `Reproduction link` in prev step

use search input

open chrome console

look error stack

### What is expected?
search is broken

### What is actually happening?
```
 TypeError: Cannot read property 'tf' of undefined
    at t.Index.documentVector (lunr.min.js:7)
    at t.Index.<anonymous> (lunr.min.js:7)
    at Array.map (<anonymous>)
    at t.SortedSet.map (lunr.min.js:7)
    at t.Index.search (lunr.min.js:7)
    at LunrSearchEngine.search (search-lunr.js:34)
    at Object.query (search-engine.js:31)
    at launchSearch (search.js:79)
    at handleUpdate (search.js:121)
    at HTMLInputElement.<anonymous> (search.js:142)
```

---
It's long time

**Comment by eperedo** [comment]
Looks like lunr.js can't handle emojis correctly.
https://github.com/olivernn/lunr.js/issues/243#issuecomment-292250180

So I think there are two solutions, either:

- Removing emojis from here https://router.vuejs.org/en/essentials/passing-props.html
[it works in my machine](http://www.giphy.com/gifs/3o7aCTjb7bjTa1oq9q)

Or

- Not include emojis in **search_index.json**

I can help with this but in case you guys want the second solution I need more information in how index are generated.

**Comment by falstack** [comment]
@eperedo sorry, i forgot this message:

when you enter chinese char, the bug will show!

**Comment by eperedo** [comment]
@falstack Not sure how to reproduce the chinese char, something like this?
![captura de pantalla 2017-10-23 a la s 22 20 53](https://user-images.githubusercontent.com/461124/31923514-8e878078-b840-11e7-9029-e8ba9a90c775.png)

**Comment by falstack** [comment]
@eperedo like this：

![kapture 2017-10-24 at 11 47 22](https://user-images.githubusercontent.com/16357724/31924083-70ad8aa8-b844-11e7-9bb7-71655916fd35.gif)

**Comment by posva** [comment]
Fixed in e5778d761d625ae0b98948f8ccaafb2ab32512e8
Will be deployed shortly
